src/mochawesome.js

```diff
@@ -81,6 +81,16 @@ function Mochawesome(runner, options) {
   // Set the config options
   this.config = conf(options);
 
+  // If the runner doesn't have a stats object then we're probably
+  // running under Mocha 6 but were started in a way that didn't
```

Comment:
I don't think this assumption is true. When running under Mocha 5, `runner.stats` will be `undefined` at this point and trying to require `stats-collector` will throw because it doesn't exist. When running directly in Mocha 6, `runner.stats` will be an object at this point.